python/lsst/obs/base/gen2to3/convertRepo.py

```diff
@@ -601,13 +639,20 @@ def run(self, root: str, *,
             pool = Pool(processes)
         if calibs is None:
             calibs = [CalibRepo(path=None)]
+        elif calibs and not self.config.doExpandDataIds:
+            raise ValueError("Cannot convert calib repos with config.doExpandDataIds=False.")
         if visits is not None:
             subset = ConversionSubset(instrument=self.instrument.getName(), visits=frozenset(visits))
         else:
             if self.config.relatedOnly:
                 self.log.warn("config.relatedOnly is True but all visits are being ingested; "
                               "no filtering will be done.")
             subset = None
+        if self.config.doExpandDataIds and self.config.transfer not in ("direct", None):
```

Comment:
As with my daf_butler comment but worse since now the transfer logic for datastore is now being assumed outside of butler.